compiler/foder/include/foder/FileLoader.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <fstream>
+#include <vector>
+
+namespace foder
+{
+
+class FileLoader
+{
+private:
+  using DataBuffer = std::vector<char>;
+
+public:
+  explicit FileLoader(const std::string &path) : _path(path) {}
+
+public:
+  FileLoader(const FileLoader &) = delete;
+  FileLoader(FileLoader &&) = delete;
+
+public:
+  const char *load(void)
+  {
+    std::ifstream file(_path, std::ios::binary | std::ios::in);
+    if (!file.good())
+      return nullptr;
+
+    file.unsetf(std::ios::skipws);
+
+    file.seekg(0, std::ios::end);
+    _size = file.tellg();
+    file.seekg(0, std::ios::beg);
+
+    // reserve capacity
+    _data.reserve(_size);
+
+    // read the data
+    file.read(_data.data(), _size);
+    if (file.fail())
+      return nullptr;
+
+    return _data.data();
+  }
```

Comment:
Do you need to keep ``DataBuffer`` insider this class? Why don't you simply return ``std::vector<char>`` and throws an exception on failure?

The run-time behavior would be almost same as this implementation due to RVO (Return-value optimization).